### ISSUE TITLE
Authentication and Authorization.

### ISSUE THREAD

**Issue by Slam256** [Slam256/task-app#2]
# Authentication and Authorization.
Currently the endpoints can be accessed by anyone and they are able to read and write or even worse delete. 
- [x] Secure the REST API adding authentication.
- [x] Protect the routes.

**Comment by Slam256** [comment]
### EPIC
- [x] As a user, I should be able to login securely, so that my task information can only be accessed by  only me.
#### User stories
- [x] As a user, no one should update/ delete / view my task, to feel secure
- [x] As a user, I should be able  to view **my task**, to feel secure
- [x] As a user, I should be able to delete **my task**, to feel in control
- [x] As a user, I should be able to update **my task**, to feel in control

**Comment by Slam256** [comment]
- I created a new collection in the database that will store user information as the goal collection stores goals.
- In order to complete the user stories above, I used [JsonWebToken](https://jwt.io/) to generate, decode and verify tokens for authentication and authorization purposes. i.e to match goals to their users. 
- I also used [bcrypt](https://en.wikipedia.org/wiki/Bcrypt) to protect user passwords against [rainbow table](https://en.wikipedia.org/wiki/Rainbow_table) attacks and I also made sure not to store their passwords in the database.
